x-pack/plugins/observability_solution/apm/server/routes/assistant_functions/get_log_categories/index.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import datemath from '@elastic/datemath';
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { getTypedSearch } from '../../diagnostics/create_typed_es_client';
+
+export type LogCategories =
+  | Array<{
+      key: string;
+      docCount: number;
+    }>
+  | undefined;
+
+export async function getLogCategories({
+  esClient,
+  arguments: args,
+}: {
+  esClient: ElasticsearchClient;
+  arguments: {
+    start: string;
+    end: string;
+  };
+}): Promise<LogCategories> {
+  const start = datemath.parse(args.start)?.valueOf()!;
+  const end = datemath.parse(args.end)?.valueOf()!;
+
+  const search = getTypedSearch(esClient);
+  const res = await search({
+    index: '*:logs-*,logs-*,*:filebeat*,filebeat*', // TODO: make this configurable somehow (UI, kibana config?)
+    size: 0,
+    track_total_hits: 0,
+    timeout: '5s',
+    query: {
+      bool: {
+        filter: [
```

Comment:
I'd like to add some kind of filter. Either on service, container or host. Just not sure which fields to filter on (it seems there is no standardisation)

Reply:
what about picking some known fields from the alert, and then adding those as a filter if they're not empty?

Reply:
Afair we decided to (almost) exclusively use ECS in alerts. Therefore I know that the host field is `host.hostname` and container fields is `container.id`. Most infrastructure rules capture these but (as I understand) this most likely won't map to how they are stored in log documents. For instance, querying logs for `host.hostname: 1234` might not give any results because it's stored as `kubernetes.node.hostname: 123`.

If you suggest we just try with ECS I'm good with that. Another approach could be to do an exact match search across all fields.

Reply:
ECS sounds good to me as a start

Reply:
Done. Added `service.name`, `container.id` and `host.name` as optional filters: https://github.com/elastic/kibana/pull/178330/commits/bccd3b9200240890f85e92c3e0b91efb0d5208c5